wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -161,11 +182,160 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
     }
   }
 
+  private case class OutputWithIndex(output: TransactionOutput, index: Int)
+
   /**
-    * @inheritdoc
+    * Processes an incoming transaction that's new to us
+    * @return `Some(tx)` if we found anything relevant to us
+    *         in the given transaction, `None` if nothing was
+    *         of relevance
+    */
+  private def processNewIncomingTx(
+      transaction: Transaction,
+      confirmations: Int): Future[Option[IncomingTransaction]] = {
+    addressDAO.findAll().flatMap { addrs =>
+      val relevantOutsWithIdx: Seq[OutputWithIndex] = {
+        val withIndex =
+          transaction.outputs.zipWithIndex
+        withIndex.collect {
+          case (out, idx)
+              if addrs.map(_.scriptPubKey).contains(out.scriptPubKey) =>
+            OutputWithIndex(out, idx)
+        }
+      }
+
+      relevantOutsWithIdx match {
+        case Nil =>
+          logger.debug(
+            s"Found no outputs relevant to us in transaction${transaction.txIdBE}")
+          Future.successful(None)
+
+        case xs =>
+          val count = xs.length
+          val outputStr = {
+            xs.map { elem =>
+                s"${transaction.txIdBE.hex}:${elem.index}"
+              }
+              .mkString(", ")
+          }
+          logger.trace(
+            s"Found $count relevant output(s) in transaction=${transaction.txIdBE}: $outputStr")
+          if (xs.length > 1) {
+            logger.warn(
+              s"${xs.length} SPKs were relevant to transaction=${transaction.txIdBE}, but we aren't able to handle more than 1 for the time being")
```

Comment:
This needs a issue at minimum, it is not uncommon for people to re-use addresses (unfortunately) and this seems to break this.